Documentation/design-docs/first-class-structs.md

```diff
@@ -346,6 +350,19 @@ static foo getfoo() { return new foo(); }
   * This is related to #3144, and requires supporting the assignment of a multi-reg call return into a promoted local variable,
     and enabling subsequent elimination of any redundant copies.
 
+* [\#22445](https://github.com/dotnet/coreclr/issues/22445) and [\#22319](https://github.com/dotnet/coreclr/issues/22319)
+  * These are both cases where we introduce a `GT_LCL_FLD` to retype a value that needs
+    to be passed in a register.
+
+## Other Struct-related Issues
+
+* [\#17207](https://github.com/dotnet/coreclr/issues/17207)
+  * This suffers from pessimization due to poor handling of conversion (`Unsafe.As`) from `Quaternion` to `Vector4`.
```

Comment:
I'll take a look. If the trees aren't broken then it may fit with may plans to handle some reinterpretation in `LocalAddressVisitor` to avoid address exposed variables. But if the trees are broken then it may be problematic, I have the forward substitution experiment but that's too late in phase order to deal with address exposed issues.

Reply:
Hmm, this may be partially related to #20957. If we delay struct promotion a bit perhaps we could avoid the seemingly unnecessary promotion that happens in this case.